Watchman.Integrations/MongoDB/Session.cs

```diff
@@ -57,6 +57,20 @@ public Session(IMongoDatabase database)
             await this.GetCollection<T>().DeleteOneAsync(x => x.Id == entity.Id);
         }
 
+        public async Task ReplaceByTypeAsync<T>(T item) where T : Entity
+        {
+            var possiblePreviousItems = this.Get<T>().ToList();
+            if (possiblePreviousItems.Count > 1)
+            {
+                throw new Exception("To replace there must be only one item of this type");
```

Comment:
czemu tylko jeden

Reply:
obiekt konfiguracja, czy też inny który korzystałby z tego rozwiązania może być tylko jeden w bazie, jakby było ich więcej (jedynie przez jakiegoś buga) wtedy dla bezpieczeństwa powinien rzucić wyjątek

Reply:
ja zrobiłbym tu takie coś
że kopiujemy kolekcje do innej kolekcji o nazwie kolekcja-copy 
albo kolekcja-backup
np `Users-backup`

usuwamy obecną kolekcje

pobieramy dane z backupowej
mapujemy na nasz nowy typ

tworzymy nową kolekcje - o nazwie takiej jak tamta usunięta
i uzupełniamy ją danymi

możliwe że mongo ma wbudowany mechanizm tego typu 
w sumie bym sie nie zdziwił
albo nasza biblioteka 

powinien być
przykładowo EF ma mechanizm migracji - który generuje zestaw komend SQL, które zmieniają strukture w bazie, żeby sie dopasować do nowej wersji kodu
przy czym możesz swobodnie sie przemieszczać po migracjach
bo każda ma w sobie mechanizm upgrade i downgrade

Reply:
żeby mieć backup ustawień kiedy coś się zepsuje?
tylko ten backup ma być jedynie ostatniego stanu?

Reply:
backup całej kolekcji w bazie

ale może mongo albo ta biblioteka do mongo udostępnia jakiś odpowiednik migracji jaki ma m.in EF